### ISSUE TITLE
pytorch backend - loss function - parameters

### ISSUE THREAD

**Issue by mchlrnwld** [voxelmorph/voxelmorph#181]
Hi all,

when using the pytorch implementation of the redesign branch, i discovered the following:
The deformation loss in pytorch_backend/losses.py is defined as 

```
class Grad:
    """
    N-D gradient loss.
    """

    def __init__(self, penalty='l1'):
        self.penalty = penalty

    def loss(self, y_true, y_pred):
        dy = torch.abs(y_true[:, :, 1:, :, :] - y_true[:, :, :-1, :, :]) 
        dx = torch.abs(y_true[:, :, :, 1:, :] - y_true[:, :, :, :-1, :]) 
        dz = torch.abs(y_true[:, :, :, :, 1:] - y_true[:, :, :, :, :-1]) 

        if self.penalty == 'l2':
            dy = dy * dy
            dx = dx * dx
            dz = dz * dz

        d = torch.mean(dx) + torch.mean(dy) + torch.mean(dz)
        return d / 3.0
```
However, the variable **y_pred** is not used in the function **loss**.
When using this loss function, shouldn't it just be applied to one variable, ie the flow field, eg
```
deformation_loss = vxm.losses.Grad('l2').loss
deformation_loss(flow)
```

Many thanks,
MR

**Comment by adalca** [comment]
@mchlrnwld 

Thank you very much for catching this. I think it's a mistake in translating to the new branch (which, while we will release very soon, will still have some quirks here and there). The current code in the pytorch folder should probably be ok.
